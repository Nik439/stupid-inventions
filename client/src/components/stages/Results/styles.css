.results-container {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.results-title {
  font-size: 10vmin;
  padding: 1.5rem;
  animation: suspense 1s ease;
}

.results-winner {
  font-size: 8vmin;
  padding: 8px;
  animation: suspense 4s ease;
}

@keyframes suspense {
  0% {
    transform: scale(0);
  }
  80% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}

.results-leaderboard {
  background-color: var(--color-primary);
  padding: 2rem;
  margin-top: 5rem;
  border-collapse: collapse;
  border-radius: 2rem;
  animation: suspense 5s ease;
}

.results-leaderboard-cell {
  font-size: 2rem;
}

.results-leaderboard-cell:first-of-type .results-leaderboard-name {
  background-color: var(--color-heading);
  border-right: 1px solid white;
  border-top: none;
  border-top-left-radius: 2rem;
}

.results-leaderboard-cell:first-of-type .results-leaderboard-votes {
  background-color: var(--color-heading);
  border-top: none;
  border-top-right-radius: 2rem;
}

.results-leaderboard-name {
  padding: 2rem;
  border-top: 1px solid white;
  border-right: 1px solid white;
}

.results-leaderboard-votes {
  padding: 2rem;
  border-top: 1px solid white;
  text-align: center;
}
