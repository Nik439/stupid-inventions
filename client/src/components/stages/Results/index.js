import React from 'react';
import './styles.css';

function Results(props) {
  const winners = props.winners.map((name, index) => {
    return (
      <h2 key={index} className="results-winner">
        {name}
      </h2>
    );
  });

  const leaderboard = props.leaderboard.map((player, index) => {
    return (
      <tr className="results-leaderboard-cell" key={index}>
        <td className="results-leaderboard-name">{player.name}</td>
        <td className="results-leaderboard-votes">{player.votes}</td>
      </tr>
    );
  });

  return (
    <div className="results-container">
      <h1 className="results-title">
        {props.winners.length > 1 ? 'Winners' : 'Winner'}
      </h1>
      {winners}

      <table className="results-leaderboard">
        <tbody data-testid="row">
          <tr className="results-leaderboard-cell">
            <th className="results-leaderboard-name">NAME</th>
            <th className="results-leaderboard-votes">VOTES</th>
          </tr>
          {leaderboard}
        </tbody>
      </table>
    </div>
  );
}

export default Results;
