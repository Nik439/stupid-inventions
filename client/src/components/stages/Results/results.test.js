import {render, screen} from '@testing-library/react';
import Results from '.';

describe('Results', () => {
  test('should render "winner" singular or plural depending on # of winners', () => {
    let winners = ['test'];
    const board = [{name: '', votes: 3}];

    const {rerender} = render(
      <Results winners={winners} leaderboard={board} />,
    );

    expect(screen.getByText('Winner')).toBeInTheDocument();
    expect(screen.queryByText('Winners')).not.toBe();

    winners = ['test', 'test2'];
    rerender(<Results winners={winners} leaderboard={board} />);

    expect(screen.getByText('Winners')).toBeInTheDocument();
    expect(screen.queryByText('Winner')).not.toBe();
  });

  test('should render winners', () => {
    let winners = ['test'];
    const board = [{name: '', votes: 3}];

    const {rerender} = render(
      <Results winners={winners} leaderboard={board} />,
    );

    expect(screen.getByText('test')).toBeInTheDocument();

    winners = ['test', 'test2'];
    rerender(<Results winners={winners} leaderboard={board} />);

    expect(screen.getByText('test')).toBeInTheDocument();
    expect(screen.getByText('test2')).toBeInTheDocument();
  });

  test('should render the leaderboard', () => {
    const winners = [''];
    const board = [
      {name: 'test', votes: 3},
      {name: 'test2', votes: 2},
    ];

    render(<Results winners={winners} leaderboard={board} />);

    expect(screen.queryAllByTestId('row')).toMatchSnapshot();
  });
});
