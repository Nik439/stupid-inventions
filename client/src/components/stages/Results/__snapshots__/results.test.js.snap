// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render the leaderboard 1`] = `
Array [
  <tbody
    data-testid="row"
  >
    <tr
      class="results-leaderboard-cell"
    >
      <th
        class="results-leaderboard-name"
      >
        NAME
      </th>
      <th
        class="results-leaderboard-votes"
      >
        VOTES
      </th>
    </tr>
    <tr
      class="results-leaderboard-cell"
    >
      <td
        class="results-leaderboard-name"
      >
        test
      </td>
      <td
        class="results-leaderboard-votes"
      >
        3
      </td>
    </tr>
    <tr
      class="results-leaderboard-cell"
    >
      <td
        class="results-leaderboard-name"
      >
        test2
      </td>
      <td
        class="results-leaderboard-votes"
      >
        2
      </td>
    </tr>
  </tbody>,
]
`;
